src/prefect/_internal/schemas/bases.py

```diff
@@ -58,7 +58,7 @@ def __eq__(self, other: Any) -> bool:
         else:
             return copy_dict == other
 
-    def __rich_repr__(self):
+    def __rich_repr__(self) -> Generator[tuple[str, Any, Any], None, None]:
```

Comment:
Pro tip: use `rich.repr.RichReprResult` for annotating these methods.

Otherwise, when only using `yield` and not other generator features like sending and returning, it's often sufficient to use `Iterator[..]` to annotate methods like these. That you used a generator function to implement the iterator is then a mere implementation detail.